_posts/2020-10-19-angular-tutorial-deutsch/2020-10-19-angular-tutorial-deutsch.md

```diff
@@ -0,0 +1,447 @@
+Um diese Eigenschaft dynamisch zu ändern, haben wir die Möglichkeit in unser Klasse selbst eine neue Property einzuführen und diese per `Property-Binding` an die Property des p-Elements zu binden.
+Hierzu setzen wir statt dem string `'true'` den Namen des Attributes in unserer Klasse auf das Binding:
+
+
+```typescript
+class InfoBoxComponent implements OnInit {
+ text = 'Additional Info-Text on our Info Box! 🎊';
+ hidden = true;
+
+ constructor() { }
+
+ ngOnInit() {
+ }
+
+}
+```
+
+```html
+ <p>info-box works!</p>
+ <p [hidden]="hidden">{{text}}</p>
+```
+
+Um die Komponente nun durch User-Interaktion zu ändern, haben wie die Möglichkeit auf sogenannte `Events` zu hören und hierfür ebenfalls ein `Event-Binding` zu definieren.
+Event-Bindings werden in Angular über Runde Klammern definiert welche den Namen des Events enthalten.
+Wenn wir nun also auf das [click Event](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event) eines eines HTML-Elements hören wollen können wir das wie folgt erreichen.
+
+   <button (click)="">
+
+Innerhalb dieser Definition haben wir nun die Möglichkeit ein sogenanntes `Template-Statement` zu definieren. Dies kann sowohl eine `Template-Expression` sein welche z.B. direkt Änderungen an Attributen eurer Klasse macht oder eine Referenz auf eine Methode in eurer Klasse.
+Um es einfach zu halten nutzen wir in diesem Fall erstmal eine `Template-Expression` welche den Wert von `hidden` jeweils negiert. Also aus `true` wird `false` und andersherum.
+
+
+```html
+ <p>info-box works!</p>
+ <button (click)="hidden=!hidden">
+ <p [hidden]="true">{{text}}</p>
+```
+
+<img class="img-fluid img-rounded" src="info-box-toggle.gif" alt="Show and hide the additional text after button click">
+
+Wir können natürlich auch jedes andere Event wie z.B. `keyup` benutzen. Mit diesem sehr simplen Mechanismus können wir generisch alle Arten von Komponenten benutzen und mit ihnen interagieren. Dies ist das unabhängig davon, ob sie in Angular oder einem anderem Framework geschrieben sind.
+
+<div class="alert alert-success">👨‍💻👩‍💻Jetzt selber nachbauen im Classroom Task: <a href="#" target="_blank">Create Info Box</a>. </div>
+
+## Schleifen mit *ngFor
+
+```html
+<ul>
+    <li *ngFor="let book of books">
+        <span>{{book.title}}</span> - <small>{{book.subtitle}}</small>
+    </li>
+</ul>
+```
+
+```typescript
+books = [
+    {
+      title: 'Book #1',
+      subtitle: 'Subtitle #1'
+    },
+    {
+      title: 'Book #2',
+      subtitle: 'Subtitle #2'
+    },
+    {
+      title: 'Book #3',
+      subtitle: 'Subtitle #3'
+    }
+  ];
+```
+
+
+<img class="img-fluid img-rounded" src="static-list-of-books.png" alt="A static list of books with the data defined in the array">
+
+<div class="alert alert-success">👨‍💻👩‍💻Jetzt selber nachbauen im Classroom Task: <a href="#" target="_blank">Show a list of books</a>. </div>
+
+## Der erste Service
+
+### Dependency Injection
```

Comment:
Ok, jetzt verstehe ich.
Der Artikel scheint noch in der Fertigung zu sein... :-D